### ISSUE TITLE
ToggleAlwaysOnBottom window rule causes workspace switch

### ISSUE THREAD

**Issue by stefonarch** [labwc/labwc#1170]
Any left or right click on the desktop of pcmanfm-qt or on lxqt-panel causes now a switch to the first desktop.
After many test it looks like the window rule `ToggleAlwaysOnBottom` is involved.

To reproduce here it's enough to have in rc.xml
```
      <windowRule identifier="qps">
      <action name="ToggleAlwaysOnBottom"/>
    </windowRule>
```
Switching to any workspace and clicking qps' window will take me to the workspace where I launched it.
The issue is present on both machines (debian/arch) with slightly different configs and it's some days I noticed it, mainly because all files clicked on the desktop open always on the first workspace, but also right click menus for the panel do so.

 If I use `ToggleAlwaysOnTop` for the panel the issue is not present anymore.

~I tried downgrading and I can reproduce it now also in 0.6.5 but for sure it wasn't noticeably at the time, which is a mistery, I agree.~

It's not reprudible that way in 0.6.5 (wrong downgrade...) and it's present already in d00f76e5627cfb303ecbb633743d7984af076db3

 Loaded also an` rc.xml `from a month ago, and a very basic one, same result.

**Comment by tsujan** [comment]
For me and *with 0.6.5*, it happens in the following way (always reproducible):

I auto-start PCManFM-Qt in its desktop mode and have

```xml
    <windowRule identifier="pcmanfm-qt" title="pcmanfm-desktop0">
      <skipTaskbar>yes</skipTaskbar>
      <skipWindowSwitcher>yes</skipWindowSwitcher>
      <action name="MoveTo" x="0" y="0" />
      <action name="ToggleAlwaysOnBottom"/>
    </windowRule>
```

in `~/.config/labwc/rc.xml`. I switch to any workspace > 1. Then I open a window of PCManFM-Qt (which shares the same process with its desktop — PCManFM-Qt is a single-process app) on that workspace. Then I close that window. After that, as soon as I click (or right click) on PCManFM-Qt's desktop, the workspace is switched to 1.

The problem won't happen here if I don't open and close a PCManFM-Qt window.

**Comment by Consolatis** [comment]
Hm.. I think this should be relatively easy to fix, I'll look into this later.
Thanks for the report.

**Comment by tsujan** [comment]
Thanks for the fast fix! Seeing that it couldn't be applied to 0.6.5, I adapted it for 0.6.5 (in `xdg_activation_handle_request()`), and the problem with PCManFM-Qt's desktop was fixed.

I think @stefonarch will soon test the original patch with the latest git source (I prefer to use the latest release for now).

**Comment by Consolatis** [comment]
Should be fixed by
- #1171

> (I prefer to use the latest release for now).

Might be a wise decision sometimes :)
